pyramus/src/main/java/fi/otavanopisto/pyramus/plugin/auth/internal/InternalAuthenticationStrategy.java

```diff
@@ -131,6 +131,26 @@ public User getUser(String username, String password) {
     }
   }
   
+  public User getUserByName(String username) {
+    UserIdentificationDAO userIdentificationDAO = DAOFactory.getInstance().getUserIdentificationDAO();
+    InternalAuthDAO internalAuthDAO = DAOFactory.getInstance().getInternalAuthDAO(); 
+    
+    InternalAuth internalAuth = internalAuthDAO.findByUsername(username);
+    if (internalAuth != null) {
+      UserIdentification userIdentification = userIdentificationDAO.findByAuthSourceAndExternalId(getName(), String.valueOf(internalAuth.getId()));
+      if(userIdentification != null){
+        
+        return userIdentification.getPerson().getDefaultUser();
+      }
+      logger.warning("UserIdentification is null");
```

Comment:
No need to log warning. Since this method is probably used to test whether a username is available, the given username not belonging to anyone is perfectly normal and simply returns null;